agreement/pseudonode.go

```diff
@@ -441,6 +441,7 @@ func (t pseudonodeVotesTask) execute(verifier *AsyncVoteVerifier, quit chan stru
 	for _, r := range verifiedResults {
 		select {
 		case t.out <- messageEvent{T: voteVerified, Input: r.message, Err: makeSerErr(r.err)}:
+			t.node.keys.RecordAsync(r.v.R.Sender, r.v.R.Round, account.Vote)
```

Comment:
Talked to Will about this in person, but I was curious why you are recording at the moment you put a vote messages on the channel, rather than adding a new for-loop over `verifiedResults` to record all the votes like the telemetry/logging reporter is doing just above. Is it try and capture a time closer to when the votes got sent to the network?

Reply:
I think either would be fine here is the previous discussion: https://github.com/algorand/go-algorand/pull/2677#discussion_r681774709

Reply:
I think it's better here since it's being done after the `t.persistStateDone` ( i.e. write to the crash db is complete ).
Ultimately, we could place it when we write to the network package, but that's likely to be an overkill.